docs/resources/shadow.md.erb

```diff
@@ -5,148 +5,255 @@ platform: linux
+      its('users') { should be_empty }
+    end
 
 ## Properties
 
+As a [plural resource](https://www.inspec.io/docs/reference/glossary/#plural_resource), all of `shadow`'s properties will return lists (that is, Ruby Arrays). `include` and `be_empty` are two of the most useful matchers you can use with lists. You can also perform manipulation of the lists, such as calling `uniq`, `sort`, `count`, `first`, `last`, `min`, and `max`.
+
 ### users
 
-The `users` property tests if the username exists `/etc/shadow`:
+A list of strings, representing the usernames matched by the filter.
 
-    its('users') { should include 'root' }
+    describe shadow
+      its('users') { should include 'root' }
+    end
 
 ### passwords
 
-The `passwords` property returns the encrypted password string from the shadow file. The returned string may not be an encrypted password, but rather a `*` or similar which indicates that direct logins are not allowed.
-
-For example:
+A list of strings, representing the encrypted password strings for entries matched by the `where` filter. Each string may not be an encrypted password, but rather a `*` or similar which indicates that direct logins are not allowed. Different operating systems use other flags here (such as `*LK*` to indicate the account is locked).
 
-    its('passwords') { should cmp '*' }
+    # Use uniq to remove duplicates, then determine
+    # if the only password left on the list is '*'
+    describe shadow.where(user: /adm$/) do
+      its('passwords.uniq.first') { should cmp '*' }
+      its('passwords.uniq.count') { should eq 1 }
+    end
 
 ### last\_changes
 
-The `last_changes` property tests the last time a password was changed:
+A list of integers, indicating the number of days since Jan 1 1970 since the password for each matching entry was changed.
 
-    its('last_changes') { should be_empty }
+    # Ensure all entries have changed their password in the last 90 days.  (Probably want a filter on that)
+    describe shadow do
+      its('last_changes.min') { should be < Date.today - 90 - Date.new(1970,1,1)   }
+    end
 
 ### min\_days
 
-The `min_days` property tests the minimum number of days a password must exist, before it may be changed:
+A list of integers reflecting the minimum number of days a password must exist, before it may be changed, for the users that matched the filter.
 
-    its('min_days') { should eq 0 }
+    # min_days seems crazy today; make sure it is zero for everyone
+    describe shadow do
+      its('min_days.uniq') { should eq [0] }
+    end
 
 ### max\_days
 
-The `max_days` property tests the maximum number of days after which a password must be changed:
+A list of integers reflecting the maximum number of days after which the password must be changed for each user matching the filter.
 
-    its('max_days') { should eq 90 }
+    # Make sure there is no policy allowing longer than 90 days
+    describe shadow do
+      its('max_days.max') { should be < 90 }
```

Comment:
If `max_days` is optional then some values might be `nil` and `max_days.max` would throw something like `ArgumentError: comparison of NilClass with 12 failed`

I guess `where { !max_days.nil? } ` is the right way to guard against that. If so, it might be better to leave the example as it currently is rather than mix the `where` concept into these straightforward examples

Reply:
So, two thoughts on this....
* `where` is not intended to be an advanced concept, or one unique to `shadow`.  All plural resources support it.  I don't want to be skittish about adding it where it makes a compelling example.
* That said, if a column can have nils, a simpler thing is to set the column to be `style: :simple`.  That calls uniq, flatten, and compact (compact discards nils).  That changes the semantics of the property a bit, but I think that's more useful here.

Reply:
Just to close the loop on the `style: :simple` question, it's not FIlterTable-wide - it's a setting we can change on a column-by-column basis. The [color property of the filtertable docs](https://github.com/inspec/inspec/blob/master/docs/dev/filtertable-usage.md#how-do-i-declare-my-interaction-with-filtertable) demonstrates it, with details [here](https://github.com/inspec/inspec/blob/master/docs/dev/filtertable-usage.md#a-colors-method-that-will-return-a-flattened-and-uniqd-array-of-values) about how its behavior changes.